### DISCUSSION TITLE
Lang::has() returns false for JSON translation if translated value is the same as original value

### DISCUSSION THREAD

**Discussion by SebastianSchoeps** [laravel/framework#39798] [Ideas]
This is the `Lang::has()' method:
```
public function has($key, $locale = null, $fallback = true)
{
    return $this->get($key, [], $locale, $fallback) !== $key;
}
```

It simply checks if the translated value is different from the original value. If no, it assumes that there is a translation.

This works well for array based translation files, but not for JSON files. E.g. "Email" could be also "Email" in German.

`Lang::has('Email', 'de')` will return `false` although the translation is contained in `de.json`. 

Is `Lang::has()` supposed to work only with array based translations?

Extending the method to the following would solve the issue:
```
public function has($key, $locale = null, $fallback = true)
{
    $locale = $locale ?: $this->locale;
    $this->load('*', '*', $locale);

    return $this->get($key, [], $locale, $fallback) !== $key || Arr::has($this->loaded['*']['*'][$locale], $key);
}
```

Thanks for your thoughts,
Sebastian

**Comment by crynobone** [comment]
This should already been fixed: https://github.com/laravel/framework/pull/47582
